llvm/lib/Target/WebAssembly/WebAssemblyISelLowering.cpp

```diff
@@ -3239,6 +3240,55 @@ static SDValue performBitcastCombine(SDNode *N,
   return SDValue();
 }
 
+static SDValue performAnyAllCombine(SDNode *N, SelectionDAG &DAG) {
+  // any_true (setcc <X>, 0, eq) => (not (all_true X))
+  // all_true (setcc <X>, 0, eq) => (not (any_true X))
+  // any_true (setcc <X>, 0, ne) => (any_true X)
+  // all_true (setcc <X>, 0, ne) => (all_true X)
+  assert(N->getOpcode() == ISD::INTRINSIC_WO_CHAIN);
+  using namespace llvm::SDPatternMatch;
+  SDLoc DL(N);
+  static auto SimdCombiner =
```

Comment:
Nit, I don't think static lambdas are that common in LLVM. Also maybe a better name for this is like `CombineSetCC`?